packages/react/src/enhancers/withServerPlaceholder.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { ComponentType } from 'react';
+import { ServerPlaceholder } from '../components/Placeholder';
+import { ComponentRendering } from '@sitecore-content-sdk/core/layout';
+import { ComponentMap } from '../components/sharedTypes';
+import { Page } from '@sitecore-content-sdk/core/client';
+
+export type ComponentProps = {
+  rendering: ComponentRendering;
+  placeholders: Record<string, React.ReactNode>;
+  componentMap?: ComponentMap;
+};
+
+export type WrapperProps = {
+  page: Page;
+  componentMap?: ComponentMap;
+  rendering: ComponentRendering;
+};
+
+export const withServerPlaceholder = <T extends ComponentProps, W extends T & WrapperProps>(
+  Component: ComponentType<T>,
+  placeholders: string[]
+) => {
+  return (props: W) => {
+    const phProps: Record<string, unknown> = {};
+    for (const placeholder of placeholders) {
+      phProps[placeholder] = (
+        <ServerPlaceholder
+          name={placeholder}
+          rendering={props.rendering}
+          page={props.page}
+          componentMap={props.componentMap}
```

Comment:
if componentMap is optional here it would throw an error.

Reply:
we'll reuse props between server and regular placeholders, so componentMap needs to be optional. I've adjusted componentMap to be madatory in HOC, but optional in other places